File: apps/auth-server/src/main/java/com/example/chat/auth/server/core/domain/Token.java
package com.example.chat.auth.server.core.domain;

import java.time.Instant;
import java.util.UUID;

/**
 * JWT 토큰 정보
 */
public record Token(
        String accessToken,
        String refreshToken,
        Instant expiresAt,
        Instant refreshExpiresAt,
        UUID principalId,
        AuthLevel authLevel,
        TokenType tokenType) {
    public boolean isExpired() {
        return Instant.now().isAfter(expiresAt);
    }
}

File: apps/auth-server/src/main/kotlin/com/example/chat/auth/server/core/domain/Token.kt
package com.example.chat.auth.server.core.domain

import java.time.Instant
import java.util.UUID

/**
 * JWT 토큰 정보
 * - 인증 성공 후 발급되는 토큰
 * - AuthLevel과 Principal 정보를 담음
 */
data class Token(
        val accessToken: String,
        val refreshToken: String?,
        val expiresAt: Instant,
        val refreshExpiresAt: Instant?,
        val principalId: UUID,
        val authLevel: AuthLevel,
        val tokenType: TokenType
) {
    fun isExpired(): Boolean = Instant.now().isAfter(expiresAt)
}
